pkg/backup/backupdest/backup_index_test.go

```diff
@@ -481,6 +483,308 @@ func TestIndexExists(t *testing.T) {
 	}
 }
 
+func TestListIndexes(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
```

Comment:
how much is this test really adding, over `TestGetBackupTreeIndexMetadata`, especially for the cases without an end time? I'd be fine deleting this.

Reply:
I will also add that i'm not the biggest fan of this test because the logic you use to construct the expected order is essentially the same as the source code logic. `TestGetBackupTreeIndexMetadata` feels like a stronger test because it has hardcoded expected values. 

All that being said, i don't feel that strongly.

Reply:
Furthermore, if we do end up deleting the filtering logic in `GetBackupTreeIndexMetadata`, then i don't see how much coverage we get from having 2 tests here. Anyways, happy to merge now and clean up later.